### PR TITLE
End to end test

### PR DESCRIPTION
**Pull request by ThomasRoehl**
End to end tests for Rocket.Chat and Smarti.

To run tests, follow this: [Rocket.Chat.Doku.Testing](https://rocket.chat/docs/developer-guides/testing/#testing)

This is a list of all the tests:

Smarti Connection]
    [Status]
      health
        ✓ Smarti should be UP
      Info
        ✓ Access Smarti info
    [Smarti Configuration]
      [Client]
        ✓ check if client already exists
        ✓ create new client
        ✓ check if right client was picked
        ✓ post query-builder
        ✓ post access token
        ✓ Login to Rocket.Chat api
        ✓ Update access token in Rocket.Chat
        ✓ Logout from Rocketchat api
    [RC Configuration]
      ✓ Login to Rocket.Chat api
      ✓ Check Assistify_AI_Smarti_Domain
      ✓ Check Assistify_AI_Source
      ✓ Check Assistify_AI_Enabled
      ✓ Check Assistify_AI_RocketChat_Webhook_Token
      ✓ Check Assistify_AI_Smarti_Base_URL
      ✓ Logout from Rocketchat api

  [Smarti Integration]
        User not logged. logging in...
    [Topic]
      Open Topic
        ✓ switch to GENERAL
        ✓ switch back to Topic
      Message
        ✓ it should send a message
    [Request]
      First request
        ✓ create is successful
        ✓ answer request
        ✓ close request
      Second request
        ✓ create is successful
        ✓ knowledgebase answer visible
        ✓ post knowledgebase answer
    Cleanup
      ✓ close new Topic

  [Smarti Cleanup]
    ✓ get Client Id
    ✓ delete client
    ✓ logout of rocket.chat 

In Travis.CI a Smarti docker image is pulled from Docker.Hub and launched during the testrun.

### CODE REVIEW

**Comment by ThomasRoehl** [comment]
The problem with the popup window is, that it is not consistently showing up (it feels somehow random). But when the popup is displayed you can not click the create before the popup is closed.
I will add a check for the popup window to fix this

File: tests/end-to-end/ui_smarti/00-preparation.js [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+			var authToken;
+			var userId;
+
+			it('check if client already exists', function (done) {
+				request.get('/client')
+					.auth(credentials['username'], credentials['password'])
+					.expect(200)
+					.expect(function (res) {
+						if (typeof res.body[0] !== 'undefined') {
+							clientid = res.body[0].id;
```
Comment (mrsimpson) [inline tests/end-to-end/ui_smarti/00-preparation.js]: @ThomasRoehl can you please make the selection of the client more robust? It was good if tests could also run with other clients present in the Smarti-server

**Comment by ThomasRoehl** [comment]
I added the tests for global announcement

**Comment by ThomasRoehl** [comment]
added tests for custom room types

**Comment by mrsimpson** [comment]
@ThomasRoehl I fixed

- real bugs found in your integration tests (settings were not properly decorated using css-selectors in the frontend)
- an error in smarti widget backend (precisely the proxy) not propagating `:` in the URL
- a bug in the rocket chat test-helpers which made it impossible to always log-in with a non-existent user

Also I added a feature to only run particular tests. Precondition: They are in a separate folder:
`CHIMP_PATH={  the folder containing the tests }` npm run chimp-path`

I'm fine with it now, feel free to merge!

**Comment by ThomasRoehl** [comment]
I just added a short readme for the tests. 
@mrsimpson can you have a quick look at this, then I will merge everything.

**Pull request merged**